### PR TITLE
PLT-7: Refactoring frontend (chunk 4)

### PR DESCRIPTION
**Pull request by enahum**
- Authorize *(.html template needs refactoring)*
- Docs
- Add `Messaging_es.md` and rename `Messaging.md to Messaging_en.md`
- Find teams
- Login
- Signup team

**Important**:

line 68 at login_email.jsx now uses err.id instead of err.message 
```javascript
if (err.id === 'api.user.login.not_verified.app_error') {
    window.location.href = '/verify_email?teamname=' + encodeURIComponent(name) + '&email=' + encodeURIComponent(email);
    return;
}
```
Meaning that validation of error messages that comes from the server should be validated against the id and not the message itself

### CODE REVIEW

File: web/static/help/Messaging_es.md [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../doc/help/Messaging_es.md
```
Comment (coreyhulen) [inline web/static/help/Messaging_es.md]: hmm.....not sure but its showing web/static/help/Messaging.md → web/static/help/Messaging_en.md renamed

Comment (enahum) [inline web/static/help/Messaging_es.md]: I renamed it cause in docs.jsx I did this:

```javascript
        if (props.site) {
            $.get(`/static/help/${props.site}_${global.window.mm_locale}.md`).then((response) => {
                this.setState({text: response});
            }, () => {
                this.setState(errorState);
            });
        } else {
            this.setState(errorState);
        }
```
So there's an easy way to show the right Messaging.md

Comment (enahum) [inline web/static/help/Messaging_es.md]: oh and the `../../../doc/help/Messaging_es.md` is because is a symlink the same as Messaging.md

Comment (coreyhulen) [inline web/static/help/Messaging_es.md]: hah...my brain read that as Messaging.md -> Messaging_es.md, makes sense now.

Comment (coreyhulen) [inline web/static/help/Messaging_es.md]: oops...wait.  anyting in docs isn't included in the build and is being moved to the mattermost/docs repo.  The files in web/static/help/ are help included with the app.  I think the one in web/static/help/Messaging_en.md needs to have the text and not be a relative link.

Comment (coreyhulen) [inline web/static/help/Messaging_es.md]: Something we'll need to fix with the build un-related to loc.  For now we should duplicate the file.

Comment (enahum) [inline web/static/help/Messaging_es.md]: Ok I can fix that for both the locales, and actually im looking at the code in web.go and the route for /docs is 'api.AppHandlerIndependent' and to use the **user locale** it should be `api.UserRequired` so I can also change that

**Comment by enahum** [comment]
ha, travis failed because it just wanted to... can you please re-run it without me having to commit anything?

**Comment by jwilander** [comment]
Re-running it

File: web/static/help/Messaging.md [hunk 2]
```diff
@@ -1 +0,0 @@
-../../../doc/help/Messaging.md
```
Comment (coreyhulen) [inline web/static/help/Messaging.md]: does this file need to be deleted now?  or is it expected to be empty?

Comment (enahum) [inline web/static/help/Messaging.md]: I didn't remove it, i just made a copy of the file. I can remove it but you said duplicate it

Comment (coreyhulen) [inline web/static/help/Messaging.md]: Sorry, yup I think remove it.  I meant i'll be duplicated in the mattermost/docs repo and in the docs dir for now.  but in web/static/help there should only be Message_en.md and _es.md  (I think!!).  Does that make sense?

Comment (enahum) [inline web/static/help/Messaging.md]: Ok I'm removing it and re-submitting

**Comment by coreyhulen** [comment]
+1

**Comment by enahum** [comment]
great bro, have another PR ready to go, I'll wait for this one to merge

**Comment by jwilander** [comment]
+1

**Pull request merged**